v2/types.go

```diff
@@ -251,6 +251,11 @@ type UpdateInstanceResponse struct {
 	// Async indicates whether the broker is handling the update request
 	// asynchronously.
 	Async bool `json:"async"`
+	// DashboardURL requires a client API version >= 2.14.
+	//
+	// DashboardURL is the URL of a web-based management user interface for
+	// the service instance.
+	DashboardURL *string `json:"dashboard_url,omitempty"`
```

Comment:
I am wondering if we should make it more clear in the client which fields are supported when. 

Like this:

```
type UpdateInstanceResponse struct {
  ... existing stuff?
  UpdateInstanceResponse2.14 `json embed` (or whatever)
}

 type UpdateInstanceResponse2.14 struct {
  DashboardURL *string `json:"dashboard_url,omitempty"`
}
```


Then the UpdateInstanceResponse becomes an assembly of new features as they roll out of the OSB and when the client is smart enough to negotiate about the version of OSB it talks, we will have an easy time picking the features apart.

Reply:
This is more of a general comment about the client code and how we articulate additions to the spec locked to versions

Reply:
@pmorie ^

Reply:
@n3wscott I agree it's starting to get confusing which features are available on which API versions. Let me think about the constructions you proposed here.

Reply:
I don't mean to block this PR, but I think we might want to do a refactoring in general.

Reply:
@n3wscott yep, that's how i understood you, want to open an issue?